satpy/readers/core/landsat.py

```diff
@@ -0,0 +1,581 @@
+        return flags
+
+    @property
+    def band_calibration(self):
+        """Return per-band saturation flag."""
+        raise NotImplementedError()
+
+    def earth_sun_distance(self):
+        """Return Earth-Sun distance."""
+        return float(self.root.find(".//IMAGE_ATTRIBUTES/EARTH_SUN_DISTANCE").text)
+
+    def build_area_def(self, bname):
+        """Build area definition from metadata."""
+        from pyresample.geometry import AreaDefinition
+
+        # Here we assume that the thermal bands have the same resolution as the reflective bands,
+        # with only the panchromatic band (b08) having a different resolution.
+        if bname in PAN_BANDLIST and self.platform_name in ["Landsat-7", "Landsat-8", "Landsat-9"]:
+            pixoff = float(self.root.find(".//PROJECTION_ATTRIBUTES/GRID_CELL_SIZE_PANCHROMATIC").text) / 2.
+            x_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/PANCHROMATIC_SAMPLES").text)
+            y_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/PANCHROMATIC_LINES").text)
+        else:
+            pixoff = float(self.root.find(".//PROJECTION_ATTRIBUTES/GRID_CELL_SIZE_REFLECTIVE").text) / 2.
+            x_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/REFLECTIVE_SAMPLES").text)
+            y_size = float(self.root.find(".//PROJECTION_ATTRIBUTES/REFLECTIVE_LINES").text)
+
+        # Get remaining geoinfo from file
+        datum = self.root.find(".//PROJECTION_ATTRIBUTES/DATUM").text
+
+        # Reading utm zone or get specific crs for arctic and antarctic
+        if self.root.find(".//PROJECTION_ATTRIBUTES/UTM_ZONE") is not None:
+            utm_zone = self.root.find(".//PROJECTION_ATTRIBUTES/UTM_ZONE").text
+            pcs_id = f"{datum} / UTM zone {utm_zone}N"
+            proj_code = f"EPSG:326{utm_zone.zfill(2)}"
+        else:
+            lat_ts = self.root.find(".//PROJECTION_ATTRIBUTES/TRUE_SCALE_LAT").text
+            if lat_ts == "-71.00000":
+                # Antarctic
+                proj_code = "EPSG:3031"
+            if lat_ts == "71.00000":
+                # Arctic
+                proj_code = "EPSG:3995"
+            pcs_id = f"{datum} / EPSG: {proj_code}N"
+
+        # We need to subtract / add half a pixel from the corner to get the correct extent (pixel centers)
+        ext_p1 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_UL_PROJECTION_X_PRODUCT").text) - pixoff
+        ext_p2 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_LR_PROJECTION_Y_PRODUCT").text) - pixoff
+        ext_p3 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_LR_PROJECTION_X_PRODUCT").text) + pixoff
+        ext_p4 = float(self.root.find(".//PROJECTION_ATTRIBUTES/CORNER_UL_PROJECTION_Y_PRODUCT").text) + pixoff
+
+        # Create area definition
+        area_extent = (ext_p1, ext_p2, ext_p3, ext_p4)
+
+        # Return the area extent
+        return AreaDefinition(f"EPSG: {proj_code}", pcs_id, pcs_id, proj_code, x_size, y_size, area_extent)
```

Comment:
@mraspaud @pnuu and anyone else who might care, what do you think about the name of the area *being* the EPSG code. I asked to make it more descriptive but now using the EPSG code in this way seems wrong. I'm not sure what a better name would be and I don't know enough about landsat to suggest something else.

Reply:
eh let's just leave it the way it is. If it ends up mattering in the future I'll make an issue or a PR.